commit 8df9b558c809840bf68b90fe137c92727ff57ec4
Author: dann frazier <dannf@hp.com>
Date:   Mon Feb 15 18:50:31 2010 -0700

    [Adjusted to apply to Debian's 2.6.18]
    commit 286e633ef0ff5bb63c07b4516665da8004966fec
    Author: Roel Kluin <roel.kluin@gmail.com>
    Date:   Wed Nov 4 08:31:59 2009 -0800
    
        isdn: hfc_usb: Fix read buffer overflow

diff --git a/drivers/isdn/hisax/hfc_usb.c b/drivers/isdn/hisax/hfc_usb.c
index b5e571a..ae7425c 100644
--- a/drivers/isdn/hisax/hfc_usb.c
+++ b/drivers/isdn/hisax/hfc_usb.c
@@ -873,8 +873,8 @@ collect_rx_frame(usb_fifo * fifo, __u8 * data, int len, int finish)
 	}
 	/* we have a complete hdlc packet */
 	if (finish) {
-		if ((!fifo->skbuff->data[fifo->skbuff->len - 1])
-		    && (fifo->skbuff->len > 3)) {
+		if (fifo->skbuff->len > 3 &&
+				!fifo->skbuff->data[fifo->skbuff->len - 1]) {
 			/* remove CRC & status */
 			skb_trim(fifo->skbuff, fifo->skbuff->len - 3);
 			if (fifon == HFCUSB_PCM_RX) {
